### PR TITLE
Let the tests remove real directory and files

### PR DESCRIPTION
**Pull request by tkdchen**
Signed-off-by: Chenxiong Qi <cqi@redhat.com>

### CODE REVIEW

File: tests/test_cachito_config.py [hunk 1]
```diff
@@ -54,11 +54,13 @@ def test_validate_cachito_config_cli(mock_isdir, app):
...
     ],
 )
-def test_validate_mutually_exclusive_package_managers(app, value, is_valid):
+def test_validate_mutually_exclusive_package_managers(app, value, is_valid, tmpdir):
     config = app.config.copy()
     config["CACHITO_MUTUALLY_EXCLUSIVE_PACKAGE_MANAGERS"] = value
 
     if is_valid:
+        # Successful validation requires this dir exists.
+        config["CACHITO_BUNDLES_DIR"] = str(tmpdir)
```
Comment (chmeliik) [inline tests/test_cachito_config.py:63]: It looks like the other tests handle this by mocking:
```python
@patch("os.path.isdir", return_value=True)
```

**Review by chmeliik — APPROVED** [review]
LGTM, one optional suggestion

File: tests/test_cachito_config.py (hunk 1 above)
Comment (tkdchen) [reply tests/test_cachito_config.py:63]: Yeah, I tend to make it explicit for this. I actually also think it should be better ensure the directory exists than mocking the isdir :) IMO, it makes it much clear what the test tests.

File: tests/test_api_v1.py [hunk 2]
```diff
@@ -1080,10 +1071,18 @@ def test_set_state_stale(
     db.session.add(request)
     db.session.commit()
 
+    bundle_dir = RequestBundleDir(1, str(tmpdir))
+    bundle_dir.mkdir(parents=True)
```
Comment (akhmanova) [inline tests/test_api_v1.py:1075]: @tkdchen Could you please explain where the test actually remove the directory? 
and why did we use mocks for that before? :)

Comment (tkdchen) [reply tests/test_api_v1.py:1075]: @akhmanova 

For the first question, https://github.com/release-engineering/cachito/blob/master/cachito/web/api_v1.py#L494

For the second one, I don't know why :)

**Pull request merged**